### ISSUE TITLE
[ray_client] Make ClientObjectRefs awaitable

### ISSUE THREAD

**Issue by edoakes** [ray-project/ray#13747]
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
```python
import asyncio

import ray

ray.util.connect("localhost:61234")

@ray.remote
def f():
    return "Hello!"

async def main():
    print(ray.get(f.remote())) # Works.
    print(await f.remote()) # Fails.

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

```
(base) code/ray % python test.py 
Hello!
Traceback (most recent call last):
  File "test.py", line 17, in <module>
    loop.run_until_complete(main())
  File "/Users/eoakes/anaconda3/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "test.py", line 14, in main
    print(await f.remote())
TypeError: object ClientObjectRef can't be used in 'await' expression
```

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by edoakes** [comment]
@AmeerHajAli this is what I was referring to that we need for serve's python API to work w/ ray client

**Comment by ijrsvt** [comment]
Closed by #15488
